src/native/index.ts

```diff
@@ -13,5 +13,5 @@ export function setupServer(
 ): SetupServerApi {
   // Provision request interception via patching the `XMLHttpRequest` class only
   // in React Native. There is no `http`/`https` modules in that environment.
-  return new SetupServerApi([XMLHttpRequestInterceptor], ...handlers)
+  return new SetupServerApi([FetchInterceptor], ...handlers)
```

Comment:
We still need support XMLHttpRequest, do we not? Has React Native officially dropped XHR as the means to make requests? If so, I'd like to see that in their changelog. 

Otherwise, we better to _add_ `FetchInterceptor` to the list here.

Reply:
React native does not have its own implementation of fetch. [It uses whatwg-fetch](https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Network/fetch.js) above their XMLHttpRequest implementation. Because we [fixed the error](https://github.com/mswjs/interceptors/pull/503) in fetch interceptor, xhr interceptor will still have the error. Is it OK to intercept both xhr and fetch if xhr will throw the error?

Reply:
Can you point me to the place where XHR interceptor would still have the error? 

The problem as in accessing the `request.signal` that doesn't always exist. But XHR interceptor doesn't access that property, afaik. Happy to be proven wrong.

Reply:
Never mind, I've just tried with both interceptors and it works

Reply:
@sairus2k, still a question from my side: can I write this code in React Native without any explicit polyfills on my end:

```js
fetch('https://example.com')
```

Reply:
Yes, fetch is the first class citizen in RN https://reactnative.dev/docs/network

Reply:
Awesome! Then what you are proposing in this pull request definitely makes sense. Let's just _add_ the fetch interceptor to the list, keeping XHR.